src/models/curriculum.ts

```diff
@@ -11,3 +13,34 @@ export const CurriculumModel = types
   });
 
 export type CurriculumModelType = typeof CurriculumModel.Type;
+
+/*
+  createFromJson
+
+  The JSON representation contains strings for things like SectionTypes.
+  CurriculumModel.create() expects proper TypeScript enumerated values instead.
+  This function recursively replaces SectionType strings with the corresponding
+  SectionType enumerated values and returns the resulting CurriculumModel.
+ */
+export function createFromJson(json: any) {
+  const snapshot = replaceSectionTypes(json);
+  return CurriculumModel.create(snapshot);
+}
+
+/*
+  replaceSectionTypes
+
+  Recursively replaces SectionType strings in 'type' fields with corresponding
+  SectionType enumerated values.
+ */
+function replaceSectionTypes(obj: {}): SnapshotIn<typeof CurriculumModel> {
+  each(obj, (v, k) => {
+    if ((k === "type") && (SectionType[v] != null)) {
+      (obj as SectionModelType).type = SectionType[v] as SectionType;
+    }
+    else if (isObject(v)) {
+      replaceSectionTypes(v);
+    }
+  });
+  return obj as SnapshotIn<typeof CurriculumModel>;
```

Comment:
Fancy! I'm not sure what `SnapshotIn` does, could you explain what's going on here?

Reply:
Just copied from the MST docs. That's the TypeScript way to identify the snapshot type, where `SnapshotIn` is the form used when creating a model object and `SnapshotOut` is the form used when serializing a model object.